### ISSUE TITLE
Windows authentication Service fabric (on-premise) application, specific username and password

### ISSUE THREAD

**Issue by Sweiii** [Azure/service-fabric-issues#1029]
Hi

I have an application with an Web API i would like to secure that with an username and password when accessing the endpoint e.g. http://localhost:8200, i have tried https://azure.microsoft.com/en-us/blog/windows-authentication-in-service-fabric-and-asp-net-core-2-0/ and it works but i can't figure out how to specify the username and password required to access the endpoint. 

Right now everyone in the domain has access but i would like to restrict it to a group.

Can someone help me with en example or better yet help me add a group to the example in the link above so i then can implement it in my own service?

Best regards,
Andreas Andersson

**Comment by masnider** [comment]
I think you may be looking for this: https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-runas-security

**Comment by Sweiii** [comment]
Thanks masnider, as I understand that link is to Run the service as a different account im looking for auth when accessing the service. e.g https://sfprod:8199 as of now every one has access to that service.

The cluster has windows as cluster security and to access https://sfprod:19080 you need to be in a specified group and thats what i want.

I need to implemen windows security on the service itself.

**Comment by Sweiii** [comment]
But i got a example from a developer that we might be able to implemen with out odata as we dont have that in our service.

Its with [Authorize(Role....)]

**Comment by RajeetN** [comment]
Yes, Runas feature is not related to this. This is service level authorization logic that SF does not have control over. Matt, @Sweiii  is probably looking for sample of how to plug principal permissions attribute in our programming model not SF runas

**Comment by masnider** [comment]
ok. Adding some people. @linggengmsft / @dragav - can someone help point where one of our existing samples would be enhanced?
